name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/dashboard/DataSeriesConfig.java

```diff
@@ -60,6 +60,18 @@ public DataSeries getDataSeries()
         return dataSeries;
     }
 
+    public void setDataSeries(DataSeries dataSeries)
+    {
+        this.dataSeries = dataSeries;
+
+        delegate.getWidget().getConfiguration().put(configurationKey.name(), dataSeries.getUUID());
+
+        delegate.onWidgetConfigEdited(this.getClass());
+
+        delegate.update();
+        delegate.getClient().touch();
+    }
```

Comment:
Only a minor thing: I changed the #doAddSeries method to also call this method to set the data series. It reduces some of the duplicated code.